modules/store/testing/src/mock_store.ts

```diff
@@ -18,9 +18,21 @@ import { MOCK_SELECTORS } from './tokens';
 if (typeof afterEach === 'function') {
   afterEach(() => {
     try {
-      const mockStore: MockStore | undefined = TestBed.inject(MockStore);
-      if (mockStore) {
-        mockStore.resetSelectors();
+      /**
+       * Since angular v13, destroyAfterEach part of the teardown option object is enabled by default
+       * which means that at this point TestBed has already been cleaned up.
+       * If destroyAfterEach is enabled, injecting MockStore in TestBed will cause an error when testing with jest
+       *
+       * More info about testing module teardown: https://dev.to/this-is-angular/improving-angular-tests-by-enabling-angular-testing-module-teardown-38kh
+       * Issue related to these changes: https://github.com/ngrx/platform/issues/3243
+       */
+      const TestBed = getTestBed() as any;
+      const shouldResetSelectors = !TestBed.shouldTearDownTestingModule();
+      if (shouldResetSelectors) {
```

Comment:
I think that because we don't reset the selectors anymore, that the selectors keep their "old" value.
What's the consequence of doing this? Should we provide another reset strategy?

Reply:
We reset the selectors every time you instantiate the mock store anyway, so I'm wondering about when we'd actually need this also.

Reply:
Seems to be related to this: 
issue: https://github.com/ngrx/platform/issues/2044
PR: https://github.com/ngrx/platform/pull/2115

And from documentation:

> Note: MockStore will reset all of the mocked selectors after each test (in the afterEach() hook) by calling the MockStore.resetSelectors() method.

Reply:
executing mock_store.spec.ts pointed me out issue related to selectors not rese  ( not sur why all test pass in CI)

Reply:
@brandonroberts we only reset the selectors that are included in a mock store instance.
With the previous implementation, that instance would be the same in the `afterEach` hook, meaning that we were we able to reset the selectors here.
With the current behavior, I don't think that `resetSelectors` is invoked (with teardown enabled), and thus that selectors won't be cleaned up in between tests.
For example, these tests are passing, while I think they should fail (and I think that these are failing on the previous version).

```ts
  it('set selectors', () => {
    store.overrideSelector(selectCollectionBookIds, [2, 3, 4] as any);
    expect(selectCollectionBookIds({})).toEqual([2, 3, 4] as any);
  });

  it('reset selectors', () => {
    // I would expect this to fail, but the selector doesn't reset in between tests
    expect(selectCollectionBookIds({})).toEqual([2, 3, 4] as any);
  });
```

Reply:
Can we can remove this hook and tackle it if it becomes a problem?
A possible solution to keep this functionality is storing the mocked selectors in a global variable, this will be the simplest solution I think.

Reply:
Yes, I'm not sure what you mean and how to handle this. Should I close this PR and let you fix this ?

Reply:
@cedricduffournet you can leave the PR open, I'm suggesting to remove the `afterEach` hook until it becomes a problem and then we can think of a solution that we're happy with. 
If @brandonroberts approves, you can remove that hook in this PR.

Reply:
I'm good with that